### PR TITLE
refactor: get video by slug API

### PR DESCRIPTION
**Pull request by csiyang**
# Description

- Refactor API to get a video by slug
- Added a test for getVideoBySlug
- Refactored video return types in video.service

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/30300650/todos/5566326330)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] Can query a video based off path

# Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged into main

### CODE REVIEW

**Comment by csiyang** [comment]
This PR requires path to be added to videoVariant first

File: apps/api-videos/src/app/modules/video/video.service.ts [hunk 1]
```diff
@@ -56,10 +73,29 @@ export class VideoService extends BaseService {
...
+          }, item.primaryLanguageId)
+          LIMIT 1 RETURN CURRENT
+        ], 0),`)
+        break
+      case IdType.slug:
+        idFilter = aql`FILTER ${playlistId} IN item.variants[*].path`
+        variantFilter.push(aql` variant: NTH(item.variants[*
+          FILTER CURRENT.path == ${playlistId}
+          LIMIT 1 RETURN CURRENT], 0),`)
+        break
```
Comment (csiyang) [inline apps/api-videos/src/app/modules/video/video.service.ts:94]: @tataihono was a bit unsure if I should add something like this for the episodes, or just remove this...

Comment (tataihono) [reply apps/api-videos/src/app/modules/video/video.service.ts:94]: path here needs to be replaced with slug.

**Review by tataihono — CHANGES_REQUESTED** [review]

File: apps/api-videos/src/app/modules/video/video.service.ts [hunk 2]
```diff
@@ -54,43 +74,43 @@ export class VideoService extends BaseService {
...
       limit = 100
     } = filter ?? {}
-    const videosView = this.db.view('videosView')
     const search = this.videoFilter(filter)
-    const idFilter =
-      idType === IdType.databaseId
-        ? aql`FILTER video._key == ${playlistId}`
-        : aql`FILTER ${playlistId} IN video.slug[*].value`
+
+    let idFilter
```
Comment (tataihono) [inline apps/api-videos/src/app/modules/video/video.service.ts:79]:
```suggestion
    let idFilter: GeneratedAqlQuery
```

**Review by tataihono — CHANGES_REQUESTED** [review]

File: apps/api-videos/src/app/modules/video/video.service.ts (hunk 2 above)
Comment (csiyang) [reply apps/api-videos/src/app/modules/video/video.service.ts:79]:
```suggestion
    let idFilter: GeneratedAqlQuery | undefined
```

File: apps/api-videos/src/app/modules/video/video.service.ts [hunk 3]
```diff
@@ -54,43 +74,43 @@ export class VideoService extends BaseService {
...
+        ], 0),`)
+        break
+      case IdType.slug:
+        idFilter = aql`FILTER ${playlistId} IN item.variants[*].slug`
+        variantFilter.push(aql` variant: NTH(item.variants[*
+          FILTER CURRENT.slug == ${playlistId}
+          LIMIT 1 RETURN CURRENT], 0),`)
+        break
+    }
+    if (idFilter == null || variantFilter == null) {
```
Comment (csiyang) [inline apps/api-videos/src/app/modules/video/video.service.ts:98]:
```suggestion
    if (idType == null || idFilter == null || variantFilter == null) {
```

File: apps/api-videos/src/app/modules/videoVariant/videoVariant.graphql [hunk 4]
```diff
@@ -17,6 +17,10 @@ type VideoVariant {
   duration: Int!
   language: Language!
   subtitle(languageId: ID, primary: Boolean): [Translation!]!
+  """
+  Address of the video
+  """
+  slug: String!
```
Comment (csiyang) [inline apps/api-videos/src/app/modules/videoVariant/videoVariant.graphql:23]: @tataihono Are we still calling this slug, or should I rename to value?

**Review by mikeallisonJS — APPROVED** [review]

**Pull request merged**